src/core/border_router/routing_manager.cpp

```diff
@@ -2035,6 +2035,20 @@ void RoutingManager::OmrPrefixManager::Evaluate(void)
         RemoveLocalFromNetData();
     }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
+    if (mFavoredPrefix.IsEmpty() || mFavoredPrefix.GetPreference() < kPdRoutePreference ||
+        mFavoredPrefix.GetPrefix() == mLocalPrefix.GetPrefix())
+    {
+        LogInfo("Trying to request prefix via DHCPv6 PD");
+        Get<RoutingManager>().mPdPrefixManager.Start();
```

Comment:
The `mPdPrefixManager` is already started/stopped by the `RoutingManager`'s `Start()` and `Stop()` methods. Starting and stopping it based on prefix availability might not be the most intuitive approach.

Instead, I suggest adding this logic directly to `PdPrefixManager`. It can check with the `OmrPrefixManager` (using its `GetFavoredPrefix` method) and then decide whether to request or release a PD prefix accordingly.

I suggest we try to keep the meaning of the `Start()` and `Stop()` methods consistent across all the nested components of the `RoutingManager`. These methods should be called when the entire `RoutingManager` is started or stopped, respectively. The `PdPrefixManager` can maintain its own internal state to track whether it needs to request or release a PD prefix.

Reply:
Added an `idle` state to it.

Also changed mIsRunning to mIsStarted for mIsPaused